lms/djangoapps/courseware/views/views.py

```diff
@@ -449,13 +450,16 @@ def register_user_access_warning_messages(request, course_key):
                 )
             )
         elif not is_enrolled and not is_staff:
+            # Only show enroll button if course is open for enrollment.
+            enroll_link = ""
+            if course_open_for_self_enrollment(course_key):
+                enroll_link = HTML('<button class="enroll-btn btn-link">{enroll_link_label}</button>').format(
```

Comment:
Note to self: check classes of other buttons.  Thought there might have been another.